server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -1723,8 +1726,33 @@ public AliasFilter buildAliasFilter(ClusterState state, String index, Set<String
     /**
      * Returns a new {@link QueryRewriteContext} with the given {@code now} provider
      */
-    public QueryRewriteContext getRewriteContext(LongSupplier nowInMillis) {
-        return new QueryRewriteContext(parserConfig, client, nowInMillis);
+    public QueryRewriteContext getRewriteContext(LongSupplier nowInMillis, IndicesRequest indicesRequest) {
+        Supplier<Map<String, IndexMetadata>> indexMetadataMapSupplier = () -> {
+            // Skip unavailable indices
+            // TODO: Support metadata lookups for indices on remote clusters
```

Comment:
There's more work than "just" supporting metadata lookups on remote clusters. The inference action for instance is not cross-cluster aware. If this is something we want to track we should open an issue instead of leaving this TODO.

Reply:
I'll remove this TODO and track CCR support through an issue